src/component.jl

```diff
@@ -1,522 +1,427 @@
+    if start_time === nothing
+        return get_initial_timestamp(ts_metadata)
     end
 
-    @assert times_remaining == 0
+    time_diff = start_time - get_initial_timestamp(ts_metadata)
+    if time_diff < Dates.Second(0)
+        throw(ArgumentError("start_time=$start_time is earlier than $(get_initial_time(ts_metadata))"))
+    end
 
-    # Run the type-specificc constructor that concatenates forecasts.
-    return forecast_type(forecasts)
-end
+    if typeof(ts_metadata) <: ForecastMetadata
+        interval = get_interval(ts_metadata)
+        if time_diff % interval != Dates.Second(0)
+            throw(ArgumentError("start_time=$start_time is not on a multiple of interval=$interval"))
+        end
+    end
 
-function _make_forecast(
-    ::Type{T},
-    component::InfrastructureSystemsComponent,
-    start_index::Int,
-    len::Int,
-    initial_time::Dates.DateTime,
-    label::AbstractString,
-) where {T <: ForecastInternal}
-    forecast = get_forecast(T, get_forecasts(component), initial_time, label)
-    ts = get_time_series(
-        _get_time_series_storage(component),
-        get_time_series_uuid(forecast);
-        index = start_index,
-        len = len,
-    )
-    return make_public_forecast(forecast, ts)
+    return start_time
 end
 
 """
-Return a TimeSeries.TimeArray where the forecast data has been multiplied by the forecasted
-component field.
+Return a time series corresponding to the given parameters.
+
+# Arguments
+- `::Type{T}`: Concrete subtype of TimeSeriesData to return
+- `component::InfrastructureSystemsComponent`: Component containing the time series
+- `name::AbstractString`: name of time series
+- `start_time::Union{Nothing, Dates.DateTime} = nothing`: If nothing, use the
+  `initial_timestamp` of the time series. If T is a subtype of Forecast then `start_time`
+  must be a multiple of the forecast interval.
+- `len::Union{Nothing, Int} = nothing`: Length in the time dimension. If nothing, use the
+  entire length.
+- `count::Int = 1`: Only applicable to subtypes of Forecast. Number of forecast windows
+  starting at `start_time` to return.
 """
-function get_forecast_values(
+function get_time_series(
     ::Type{T},
-    mod::Module,
-    component::InfrastructureSystemsComponent,
-    initial_time::Dates.DateTime,
-    label::AbstractString,
-) where {T <: Forecast}
-    forecast = get_forecast(T, component, initial_time, label)
-    return get_forecast_values(mod, component, forecast)
-end
-
-function get_forecast_values(
-    mod::Module,
     component::InfrastructureSystemsComponent,
-    forecast::Forecast,
-)
-    scaling_factors = get_data(forecast)
-    label = get_label(forecast)
-    accessor_func = getfield(mod, Symbol(label))
-    data = scaling_factors .* accessor_func(component)
-    return data
-end
+    name::AbstractString;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+    count::Int = 1,
```

Comment:
I'm not sure that this function has ideal default behavior.  If T is a Forecast then it returns only one forecast window.  This could be non-intuitive for users exploring data.  Since the function returns the Forecast object I think it would be better to include all data by default.  If you're concerned about overflowing users' memory, we could put some limit.

The same point applies to all `get_time_series_*` functions.

Reply:
Discussion from today:  Default behavior for `get_time_series` (only) is to return all data.  In the Hdf storage file we will add a print if the size being read from disk into memory is greater than something like 100 MiB.